rotkehlchen/data_import/importers/binance.py

```diff
@@ -315,40 +317,43 @@ def process_trades(
                     )
                 continue
 
-            rate = to_amount / from_amount
-            trade = Trade(
-                timestamp=timestamp,
+            swap_events = create_swap_events(
+                timestamp=ts_sec_to_ms(timestamp),
                 location=Location.BINANCE,
-                trade_type=trade_type,
-                base_asset=to_asset,
-                quote_asset=from_asset,
-                amount=to_amount,
-                rate=Price(rate),
-                fee_currency=fee_asset,
-                fee=fee_amount,
-                link='',
-                notes='Imported from binance CSV file. Binance operation: Buy / Sell',
+                spend_asset=from_asset,
+                spend_amount=from_amount,
+                receive_asset=to_asset,
+                receive_amount=to_amount,
+                fee_amount=fee_amount,  # type: ignore[arg-type]
+                fee_asset=fee_asset,  # type: ignore[arg-type]
+                event_identifier=f'{EVENT_IDENTIFIER_PREFIX}{hash_binance_csv_row(trade_rows[0])}',  # any row works, just using the first to create the event identifier to aid the grouping below  # noqa: E501
+                spend_notes='Imported from binance CSV file. Binance operation: Buy / Sell',
             )
-            raw_trades.append(trade)
-
-        # Sometimes we can get absolutely identical trades (including timestamp) but the database
-        # allows us to add only one of them. So we combine these trades into a huge single trade
-        # First step: group trades
-        grouped_trades: dict[TradeID, list[Trade]] = defaultdict(list)
-        for trade in raw_trades:
-            grouped_trades[trade.identifier].append(trade)
+            for event in swap_events:
+                grouped_events[event.event_identifier].append(event)
```

Comment:
```suggestion
            grouped_events[event.event_identifier].extend(swap_events)
```
No need to loop here. Actually don't even use the swap_events variable - put `create_swap_events` inside the extend.

Reply:
sorry, fixed it